src/rewrite/steward.ts

```diff
@@ -190,24 +264,45 @@ export function handleLogBuy(event: LogBuy): void {
   if (isVintageVitalik(tokenIdBigInt, event.block.number)) {
     // BE VERY CAREFUL HERE, there was an issue upgrading vitalik that we must take into account.
     // This was the transaction that simon upgraded vitalik (so the deposit was updated!)
-    // if (
-    //   event.transaction.hash.toHexString() !=
-    //   "0x819abe91008e8e22034b57efcff070c26690cbf55b7640bea6f93ffc26184d90"
-    // ) {
+    // Over here do all the accounting necessary for the upgrade.
+    if (
+      event.transaction.hash.toHexString() !=
+      "0x819abe91008e8e22034b57efcff070c26690cbf55b7640bea6f93ffc26184d90"
+    ) {
+      let newPatronTokenCostScaledNumerator = VITALIK_PATRONAGE_NUMERATOR.times(
+        VITALIK_PRICE_WHEN_OWNED_BY_SIMON
+      );
+      let newPatronTotalContributed = patron.totalContributed.plus(
+        newPatronTokenCostScaledNumerator
+          .times(timeSinceLastUpdatePatron)
+          .div(GLOBAL_PATRONAGE_DENOMINATOR)
+          .div(NUM_SECONDS_IN_YEAR_BIG_INT)
+      );
+      patron.lastUpdated = txTimestamp;
+      patron.totalTimeHeld = newPatronTotalTimeHeld;
+      patron.tokens = newPatronTokenArray;
+      patron.patronTokenCostScaledNumerator = newPatronTokenCostScaledNumerator;
+      patron.totalContributed = newPatronTotalContributed;
+
+      patron.save();
+    }
     return;
```

Comment:
This return should prevent the `<entity>.save()`s below from happening on this tx to fix vitalik (https://etherscan.io/tx/0x8961f14a43c82842d5541da454b44aad64cbbe11b2f9094d6dc191bce3eac2f3), but it doesn't for some weird reason. I couldn't work out why :/

Reply:
In general though, I think it should exclusively be the duty of "phase 3" to call the `save()` function in any case. It is out of place to call it here.